src/clients/Client.ts

```diff
@@ -1,8 +1,55 @@
+import { Snowflake } from 'discord-api-types';
 import merge from 'lodash.merge';
+import EventEmitter from 'node:events';
+import { CloseEvent } from 'ws';
 import { WebSocketManager } from '../gateway/WebSocketManager';
+import { Events } from '../utils/events';
 import { DeepRequired } from '../utils/types';
 import { BaseClient, BaseClientOptions } from './BaseClient';
 
+interface ClientEvents extends Record<Events, readonly unknown[]> {
+	ready: [];
+	websocketDisconnecting: [CloseEvent];
```

Comment:
```suggestion
	websocketDisconnect: [CloseEvent];
```
it's maybe a better name ?
(`gatewayDisconnect` ?)
same for others